segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/DefragmentOperation.java

```diff
@@ -144,11 +147,28 @@
                             CompletableFuture<Void> f;
                             if (chunksToConcat.size() > 1) {
                                 // Concat
-                                f = concatChunks();
+                                f = concatChunks()
+                                .handleAsync((vv, ex) -> {
+                                    if (null != ex) {
+                                        ex = Exceptions.unwrap(ex);
+                                        if (ex instanceof InvalidOffsetException) {
```

Comment:
- We accidently encountered this issue while running longevity runs with wrong config. 
    - In SLTS concat operation is 3 step process - first we merge the chunk metadata list ,  then we defragment the chunks in that list and then finally commit all metadata changes.
    -  In this particular instance, the defragment operation was successful but the transaction to commit metadata failed. (because of internal error with table segment code).
    - This resulted in last chunk of target segment having all data from chunks of source segment but metadata still had old information.
    - When storage writer attempted the operation again, we found that for the last chunk of target segment the length on LTS did not match metadata and it correctly threw InvalidOffsetException
    - This results in unrecoverable error. SW can not make any progress and cache becomes full and everything comes to halt.
- Note that we DO NOT delete source chunks unless commit was successful.

- However this is a more generic failure mode and can happen for any of following reason
    - process crashed while doing defrag
    - some I/O error happened while doing defrag
    - the segment store failed over and the zombie instance continued to write to last chunk. (perfectly valid scenario for SLTS)
    - so on.
- Such kind of error is not data corruption - this is a "normal" and valid circumstance .
    - The appropriate handling is to skip over the chunk with garbage data at the end.
    - This kind of situation should never block StorageWriter from making progress.